.github/workflows/checks.yml

```diff
@@ -0,0 +1,35 @@
+name: Checks
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  check-links:
+    runs-on: ubuntu-latest
```

Comment:
I took this pattern from the `build.yaml` workflow.  If we change it here we should change it there as well.  Want me to make that change in this PR?

Reply:
It would be good to know if the build is suddenly breaking when the dependencies update, because we should react to that and fix our build shouldn't we? But I'm curious how this would break

Reply:
>  But I'm curious how this would break

New versions of Ubuntu can use newer major version of `clang llvm libbpf-dev` and these can cause build failures.

> because we should react to that and fix our build shouldn't we?

Of course, but if everything is "auto-updated" then it is harder to find the root cause of the issue.